core/user_api.php

```diff
@@ -535,9 +535,11 @@ function user_create( $p_username, $p_password, $p_email = '',
 
 	user_ensure_name_valid( $p_username );
 	user_ensure_name_unique( $p_username );
-	user_ensure_email_unique( $p_email );
-	email_ensure_valid( $p_email );
-	email_ensure_not_disposable( $p_email );
+	if ( ! is_blank($p_email) ) {
```

Comment:
Proper formatting is `if( !is_blank( $p_email ) ) {`.  But why do we need this check?  The existing methods should apply the right validation based on configuration like whether or not to allow empty email addresses via `$g_allow_blank_email`.

Reply:
I've added this to make the autoprov work. I've failed to notice the $g_allow_blank_email in the code, but there can be scenarios, when it's not possible to turn it on (=allowing blank e-mails); like if signup is enabled, but I want to use external auth besides it with autoprov.

My initial problem was, than if I supply an e-mail to user_create, than a signup e-mail will be sent to the user, which I want to avoid. I would also be possible to add a new parameter to the function, like 'p_dont_send_signup_email', but I did not want to do that not knowing your future ideas with user_create.